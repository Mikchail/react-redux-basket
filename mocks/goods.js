const goods = [{
  "id": 2,
  "img" : "https://img.favpng.com/2/4/5/steel-guitar-acoustic-guitar-electric-guitar-bass-guitar-png-favpng-MYXXnrv6tC6Hg2T1k7zKjW8pr.jpg",
  "name" : "Fender",
  "price" : 60000,
  "quantity" : 22
},
{
  "id": 22,
  "img" : "https://img95.699pic.com/element/40135/1104.png_860.png",
  "name" : "Fender",
  "price" : 22000,
  "quantity" : 2
},
{
  "id": 123,
  "img" : "https://img.favpng.com/20/11/20/bass-guitar-electric-guitar-acoustic-guitar-supro-ozark-1560-s-png-favpng-Auv04m1w16rLVMr6RZn433D7P_t.jpg",
  "name" : "Fender",
  "price" : 14000,
  "quantity" : 22
},
{
  "id": 1212,
  "img" : "https://img.favpng.com/22/25/5/bass-guitar-art-string-instruments-drawing-png-favpng-NEmJB8UcuuuUiN1ipbpHEJYCF.jpg",
  "name" : "Fender",
  "price" : 12000,
  "quantity" : 12
},
{
  "id": 2009,
  "img" : "https://img.pngio.com/download-electric-guitar-png-images-background-transparent-bass-hd-bass-guitar-png-840_767.png",
  "name" : "Fender",
  "price" : 15000,
  "quantity" : 2
}
]

export {goods};