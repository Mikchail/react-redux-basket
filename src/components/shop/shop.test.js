import React from 'react';
import renderer from 'react-test-renderer';
import {Provider} from 'react-redux';
import configureStore from 'redux-mock-store';

import NameSpace from '../../store/name-space';
import {goods} from '../../../mocks/goods';
import Shop from './shop';

const mockStore = configureStore([]);

describe(`Testing shop component`, () => {
  const basker = []
  const store = mockStore({
    [NameSpace.DATA]: {
      goods: goods,
      isLoading: true,
      basket: [],
      totalCount: 0,
    },
  });

  it(`render shop with data`, () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Shop basket={basker} />
        </Provider>
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
