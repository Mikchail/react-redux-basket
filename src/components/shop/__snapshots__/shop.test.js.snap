// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing shop component render shop with data 1`] = `
<ul
  className="card-list"
>
  <li
    className="card"
  >
    <h4
      className="card__title"
    >
      Fender
    </h4>
    <img
      alt="Fender"
      className="card__img"
      height="400"
      src="https://img.favpng.com/2/4/5/steel-guitar-acoustic-guitar-electric-guitar-bass-guitar-png-favpng-MYXXnrv6tC6Hg2T1k7zKjW8pr.jpg"
      width="300"
    />
    <p
      className="card__quantity"
    >
      <span
        className="span"
      >
        Quantity: 
      </span>
      22
    </p>
    <p
      className="card__price"
    >
      <span
        className="span"
      >
        Price: 
      </span>
      60000
    </p>
    <button
      className="button card__button"
      onClick={[Function]}
    >
      Add to basket
    </button>
  </li>
  <li
    className="card"
  >
    <h4
      className="card__title"
    >
      Fender
    </h4>
    <img
      alt="Fender"
      className="card__img"
      height="400"
      src="https://img95.699pic.com/element/40135/1104.png_860.png"
      width="300"
    />
    <p
      className="card__quantity"
    >
      <span
        className="span"
      >
        Quantity: 
      </span>
      2
    </p>
    <p
      className="card__price"
    >
      <span
        className="span"
      >
        Price: 
      </span>
      22000
    </p>
    <button
      className="button card__button"
      onClick={[Function]}
    >
      Add to basket
    </button>
  </li>
  <li
    className="card"
  >
    <h4
      className="card__title"
    >
      Fender
    </h4>
    <img
      alt="Fender"
      className="card__img"
      height="400"
      src="https://img.favpng.com/20/11/20/bass-guitar-electric-guitar-acoustic-guitar-supro-ozark-1560-s-png-favpng-Auv04m1w16rLVMr6RZn433D7P_t.jpg"
      width="300"
    />
    <p
      className="card__quantity"
    >
      <span
        className="span"
      >
        Quantity: 
      </span>
      22
    </p>
    <p
      className="card__price"
    >
      <span
        className="span"
      >
        Price: 
      </span>
      14000
    </p>
    <button
      className="button card__button"
      onClick={[Function]}
    >
      Add to basket
    </button>
  </li>
  <li
    className="card"
  >
    <h4
      className="card__title"
    >
      Fender
    </h4>
    <img
      alt="Fender"
      className="card__img"
      height="400"
      src="https://img.favpng.com/22/25/5/bass-guitar-art-string-instruments-drawing-png-favpng-NEmJB8UcuuuUiN1ipbpHEJYCF.jpg"
      width="300"
    />
    <p
      className="card__quantity"
    >
      <span
        className="span"
      >
        Quantity: 
      </span>
      12
    </p>
    <p
      className="card__price"
    >
      <span
        className="span"
      >
        Price: 
      </span>
      12000
    </p>
    <button
      className="button card__button"
      onClick={[Function]}
    >
      Add to basket
    </button>
  </li>
  <li
    className="card"
  >
    <h4
      className="card__title"
    >
      Fender
    </h4>
    <img
      alt="Fender"
      className="card__img"
      height="400"
      src="https://img.pngio.com/download-electric-guitar-png-images-background-transparent-bass-hd-bass-guitar-png-840_767.png"
      width="300"
    />
    <p
      className="card__quantity"
    >
      <span
        className="span"
      >
        Quantity: 
      </span>
      2
    </p>
    <p
      className="card__price"
    >
      <span
        className="span"
      >
        Price: 
      </span>
      15000
    </p>
    <button
      className="button card__button"
      onClick={[Function]}
    >
      Add to basket
    </button>
  </li>
</ul>
`;
